arangod/RestServer/DatabaseFeature.cpp

```diff
@@ -146,82 +130,54 @@ void DatabaseManagerThread::run() {
-            if (database->isOwnAppsDirectory() && !appPath.empty()) {
-              std::lock_guard lockCreate{databaseFeature._databaseCreateLock};
-
-              // but only if nobody re-created a database with the same name!
-              std::lock_guard lock{databaseFeature._databasesMutex};
-
-              TRI_vocbase_t* newInstance =
-                  databaseFeature.lookupDatabase(database->name());
-              TRI_ASSERT(newInstance == nullptr ||
-                         newInstance->id() != database->id());
-              if (newInstance == nullptr) {
-                std::string const dirName = ::getDatabaseDirName(
-                    database->name(), std::to_string(database->id()));
-                std::string path = arangodb::basics::FileUtils::buildFilename(
-                    arangodb::basics::FileUtils::buildFilename(appPath, "_db"),
-                    dirName);
-
-                if (TRI_IsDirectory(path.c_str())) {
-                  LOG_TOPIC("041b1", TRACE, arangodb::Logger::FIXME)
-                      << "removing app directory '" << path << "' of database '"
-                      << database->name() << "'";
-
-                  TRI_RemoveDirectory(path.c_str());
-                }
-              }
-            }
-          }
+        TRI_ASSERT(!database->isSystem());
+        if (ServerState::instance()->isCoordinator()) {
+          continue;  // TODO(MBkkt) Why do we have this thread on Coordinator?
+        }
 
-          auto shutdownRes = basics::catchVoidToResult(
-              [&database]() { database->shutdown(); });
-          if (shutdownRes.fail()) {
-            LOG_TOPIC("b3db4", ERR, Logger::FIXME)
-                << "failed to shutdown database '" << database->name()
-                << "': " << shutdownRes.errorMessage();
-          }
+        auto r = basics::catchVoidToResult([&]() { database->shutdown(); });
+        if (!r.ok()) {
+          LOG_TOPIC("b3db4", ERR, Logger::FIXME)
+              << "failed to shutdown database '" << database->name()
+              << "': " << r.errorMessage();
+        }
 
-          // destroy all items in the QueryRegistry for this database
-          auto queryRegistry = QueryRegistryFeature::registry();
-          if (queryRegistry != nullptr) {
-            // but only if nobody re-created a database with the same name!
-            std::lock_guard lock{databaseFeature._databasesMutex};
-            TRI_vocbase_t* newInstance =
-                databaseFeature.lookupDatabase(database->name());
-            TRI_ASSERT(newInstance == nullptr ||
-                       newInstance->id() != database->id());
-
-            if (newInstance == nullptr) {
+        search.removeDatabase(*database);
+
+        auto* queryRegistry = QueryRegistryFeature::registry();
+        if (dealer.isEnabled() || queryRegistry != nullptr) {
+          // TODO(MBkkt) Why shouldn't we remove database data
+          //  if exists database with same name?
```

Comment:
Because this thread executes the deletion at some point _after_ the user issued a `db._dropDatabase()` call.
Consider the following situation:
- user executes `db._dropDatabase("foo");`
- user executes `db._createDatabase("foo");`
- DatabaseManagerThread now finds the deleted database "foo". Shall it remove its application directory now? We cannot be sure about it, because it may be the new database's application directory.

Reply:
I understand, but we use id for new databases in path for v8 stuff (query register still use name :()

Reply:
Not always, please check following code

```cpp
/// @brief return either the name of the database to be used as a folder name,
/// or its id if its name contains special characters and is not fully supported
/// in every OS
[[nodiscard]] static std::string_view getDatabaseDirName(std::string_view name,
                                                         std::string_view id) {
  bool const isOldStyleName = DatabaseNameValidator::isAllowedName(
      /*allowSystem=*/true, /*extendedNames=*/false, name);
  return (isOldStyleName || id.empty()) ? name : id;
}
```